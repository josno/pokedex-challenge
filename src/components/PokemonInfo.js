import React, { useState, useEffect } from "react";

const PokemonInfo = (props) => {
	const [currentPokemon, setCurrentPokemon] = useState({
		name: "",
		weight: "",
		height: "",
		imgUrl: "",
	});

	useEffect(() => {
		fetch(`https://pokeapi.co/api/v2/pokemon/${props.pokemonId}`)
			.then((response) => response.json())
			.then((responseJson) =>
				setCurrentPokemon({
					name: responseJson.name,
					weight: responseJson.weight,
					height: responseJson.height,
					imgUrl: responseJson.sprites.front_default,
				})
			);
	}, [props.pokemonId]);
	return (
		<div>
			<div>
				<img
					src={currentPokemon.imgUrl}
					alt={`${currentPokemon.name} Default`}
				/>
				<h1>{currentPokemon.name}</h1>
				<ul>
					<li>Weight: {currentPokemon.weight}</li>
					<li>Height: {currentPokemon.height}</li>
				</ul>
			</div>
		</div>
	);
};

export default PokemonInfo;
