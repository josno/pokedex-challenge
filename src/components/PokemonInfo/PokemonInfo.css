.pokemon-details-container ul {
	list-style: none;
	padding-left: 0px;
}

.pokemon-details-container img {
	width: 100%;
}

.pokemon-details-container h1 {
	margin: 0px;
}

.pokemon-details-wrapper {
	bottom: 10%;
	position: relative;
}

.pokemon-details-container {
	align-items: center;
	background-color: #262927;
	border: 1px solid black;
	border-radius: 5px;
	display: flex;
	flex-direction: column;
	height: 250px;
	justify-content: center;
	margin: 15px;
	max-width: 400px;
}

.details-text {
	color: #3ee052;
	margin: 0px;
}

.outer-details-container {
	background-color: #c4c4c4;
	/* border: 1px solid black; */
	border-radius: 10px 10px 10px 70px;
	box-shadow: 6px 6px #6b6868;
	margin: 20px 20px;
	padding: 20px 20px;
}

.image-container {
	align-items: center;
	display: flex;
	justify-content: center;
	overflow: hidden;
	width: 150px;
	height: 150px;
}
