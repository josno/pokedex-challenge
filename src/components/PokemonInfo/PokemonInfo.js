import React, { useState, useEffect } from "react";
import "./PokemonInfo.css";

const PokemonInfo = ({ listError, pokeUrl }) => {
	const [currentPokemon, setCurrentPokemon] = useState({
		name: "",
		weight: "",
		height: "",
		imgUrl: "",
	});
	const [loading, setLoading] = useState(true);
	const [loadingMessage, setLoadingMessage] = useState("Loading...");

	useEffect(() => {
		if (listError) {
			setLoadingMessage("");
			setLoading(true);
		} else {
			fetch(`${pokeUrl}`)
				.then((response) => response.json())
				.then((responseJson) => {
					const capitalizedName =
						responseJson.name.charAt(0).toUpperCase() +
						responseJson.name.slice(1);
					setCurrentPokemon({
						id: responseJson.id,
						name: capitalizedName,
						weight: responseJson.weight,
						height: responseJson.height,
						imgUrl: responseJson.sprites.front_default,
					});
					setLoading(false);
				})
				.catch((err) => {
					setLoadingMessage("Pokemon unavailable.");
				});
		}
	}, [listError, pokeUrl]);

	return (
		<section className='pokemon-details-container'>
			{loading ? (
				<div className='loading-text details-text'>{loadingMessage}</div>
			) : (
				<>
					<div className='image-container'>
						<img
							className='poke-image'
							src={currentPokemon.imgUrl}
							alt={`${currentPokemon.name} Default`}
						/>
					</div>

					<h3 className='details-text'>
						{`#${currentPokemon.id}`} <br />
						{currentPokemon.name}
					</h3>
					<ul className='details-text'>
						<li>Weight: {currentPokemon.weight}</li>
						<li>Height: {currentPokemon.height}</li>
					</ul>
				</>
			)}
		</section>
	);
};

export default PokemonInfo;
