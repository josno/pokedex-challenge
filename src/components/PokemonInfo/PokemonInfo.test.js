import React from "react";
import ReactDOM from "react-dom";

import PokemonInfo from "./PokemonInfo";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("PokemonInfo", () => {
	it(`renders without crashing`, () => {
		const div = document.createElement("div");
		ReactDOM.render(<PokemonInfo />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it(`renders the UI as expected`, () => {
		const wrapper = mount(<PokemonInfo />);
		expect(wrapper.find("div").props().children).toBe("Loading...");
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

describe.only("PokemonInfo - Testing Hooks", () => {
	let container = null;
	beforeEach(() => {
		// setup a DOM element as a render target
		container = document.createElement("div");
		document.body.appendChild(container);
	});

	afterEach(() => {
		// cleanup on exiting
		unmountComponentAtNode(container);
		container.remove();
		container = null;
	});

	it("renders pokemon info", async () => {
		const pokemon = {
			id: 25,
			name: "pikachu",
			weight: 60,
			height: 4,
			sprites: {
				front_default:
					"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
			},
		};

		jest.spyOn(global, "fetch").mockImplementation(() =>
			Promise.resolve({
				json: () => Promise.resolve(pokemon),
			})
		);

		const testUrlProp = "https://pokeapi.co/api/v2/pokemon/25";

		await act(async () => {
			render(<PokemonInfo pokeUrl={testUrlProp} />, container);
		});

		expect(container.querySelector("h3").textContent).toBe(
			`#${pokemon.id.toString()}`
		);
		expect(container.querySelector("ul").textContent).toContain(
			`Weight: ${pokemon.weight / 10} kg`
		);
		expect(container.querySelector("ul").textContent).toContain(
			`Height: ${pokemon.height / 10} m`
		);
		expect(container.querySelector("ul").textContent).toContain(
			`Name: ${pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}`
		);
		// expect(container.querySelector("img").textContent).toBe(
		// 	"https://pokeapi.co/api/v2/pokemon/25"
		// );
		expect(container.querySelector("img").src).toBe(
			pokemon.sprites.front_default
		);
	});
});
