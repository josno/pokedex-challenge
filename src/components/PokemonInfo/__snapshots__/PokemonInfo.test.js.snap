// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonInfo  renders the UI as expected 1`] = `
<PokemonInfo>
  <section
    className="pokemon-details-container"
  >
    <div
      className="loading-text details-text"
    >
      Loading...
    </div>
  </section>
</PokemonInfo>
`;
